integration-tests/tests/advanced-happy-path.spec.ts

```diff
@@ -0,0 +1,171 @@
+        }
+      });
+    });
+  });
+
+  describe('Explore "Latest commits" Tab', () => {
+    it("Verify the Commits List view", () => {
+      Applications.goToLatestCommitsTab();
+
+      for (let commitObject of latestCommitsTabPage.commitList) {
+        Common.checkRowValues(commitObject.sha, [commitObject.title, "main", componentName, commitObject.user, "Succeeded"]);
+      }
+    });
+  });
+
+  describe('Check Component Deployment', () => {
+    it("Verify the status code and response body of the deployment URL of each component", () => {
+      Applications.goToComponentsTab();
+
+      applicationDetailPage.expandDetails(componentName);
+
+      cy.get(applicationDetailPagePO.route.replace('{0}', componentName), { timeout: 35000 }).invoke('text').then(route => {
+        cy.wait(20000);
+        cy.request({
+          url: route,
+          retryOnStatusCodeFailure: true
+        }).then((response) => {
+          expect(response.status).to.eq(200);
+          expect(response.body).to.include(componentInfo.get('deploymentBodyOriginal'));
+        })
+      });
+
+      Applications.checkComponentStatus(componentName, 'Build Succeeded');
+    })
+  });
+
+  describe('Add a new commit and verify changes in the Route', () => {
+    it("Add a new commit with changes to a file", () => {
+      latestCommitsTabPage.editFile(
+        publicRepos[0], componentInfo.get('filePath'),
+        componentInfo.get('commitMessageUpdate'),
+        componentInfo.get('goFileBase64Updated'),
+        componentInfo.get('goFileSHAOriginal')
+      );
+    });
+
+    it("Check the new Pipeline run", () => {
+      Applications.goToActivityTab();
+      Applications.goToPipelinerunsTab();
+
+      cy.wait(10000);
+      cy.get('tbody tr', { timeout: 10000 }).eq(0).then(($row) => {
+        cy.wrap($row).find('td').eq(3).find('span').eq(1).then((pipelinerunStatus) => {
+          assert.equal(pipelinerunStatus.text().trim(), "Running");
+        });
+        cy.wrap($row).find('td').eq(0).find('a').then((pipelinerunName) => {
+          Applications.createdPipelinerunSucceeded(pipelinerunName.text().trim());
+          pipelinerunsTab.pipelineRunList.push(pipelinerunName.text().trim());
+        });
+      });
+    });
+
+    it("Verify the Commits List view", () => {
+      Applications.goToLatestCommitsTab();
+
+      for (let commitObject of latestCommitsTabPage.commitList) {
+        if (commitObject.title == componentInfo.get('commitMessageUpdate')) {
+          Common.checkRowValues(commitObject.sha, [commitObject.title, "main", componentName, commitObject.user, "Succeeded"]);
+        }
+      }
+    });
+
+    it("Verify that the component deployment reflects latest changes", () => {
+      Applications.goToComponentsTab();
+
+      applicationDetailPage.expandDetails(componentName);
+
+      cy.get(applicationDetailPagePO.route.replace('{0}', componentName), { timeout: 35000 }).invoke('text').then(route => {
+        cy.wait(240000);
```

Comment:
we don't like long unconditional waits here

Reply:
done, thanks